core/kernel/semihosting.c

```diff
@@ -0,0 +1,129 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright (c) 2024 Andes Technology Corporation
+ */
+
+#include <kernel/semihosting.h>
+#include <string.h>
+
```

Comment:
Please move the comment from `config.mk` to here:
```
/* 
 * ARM and RISC-V have defined the standard way to perform
 * the semihosting operations.
 * - Operation codes and open modes are identical.
 * - The implementation of the low-level __do_semihosting() call is architecture-specific.
 * - Arm semihosting interface:
 *   https://developer.arm.com/documentation/dui0471/g/Semihosting/The-semihosting-interface
 * - RISC-V semihosting interface:
 *   https://github.com/riscv-non-isa/riscv-semihosting/blob/main/binary-interface.adoc
 */
```

Reply:
Done